src/shared/Microsoft.Git.CredentialManager/GitConfiguration.cs

```diff
@@ -128,12 +128,63 @@ public void Enumerate(GitConfigurationEnumerationCallback cb)
                         throw CreateGitException(git, "Failed to enumerate all Git configuration entries");
                 }
 
-                IEnumerable<string> entries = data.Split('\0').Where(x => !string.IsNullOrWhiteSpace(x));
-                foreach (string entry in entries)
+                var scope = new StringBuilder();
+                var name  = new StringBuilder();
+                var value = new StringBuilder();
+                int i = 0;
+                while (i < data.Length)
                 {
-                    string[] kvp = entry.Split(new[]{'\n'}, count: 2);
+                    scope.Clear();
+                    name.Clear();
+                    value.Clear();
+
+                    // Read config scope (null terminated)
+                    while (data[i] != '\0')
+                    {
+                        scope.Append(data[i++]);
+                    }
+
+                    // Skip the null terminator
+                    i++;
+
+                    // Read key name (LF terminated)
+                    while (data[i] != '\n')
```

Comment:
How much should we be protecting against malformed data? Should we be `throw`ing if `i >= data.length`?

This feels like a pedantic complaint, but I'd hate to track down an `IndexOutOfBoundsException` instead of  a message saying "Output was malformed".

Reply:
@derrickstolee I've added one more commit to check for unexpected end of data and to trace and break from the parsing loop.